rflx/pyrflx/typevalue.py

```diff
@@ -716,6 +730,53 @@ def _preset_fields(self, fld: str) -> None:
             self._last_field = nxt
             nxt = self._next_field(nxt)
 
+    def set_checksum_function(self, checksum_field_name: str, checksum_method: Callable) -> None:
+        if checksum_field_name not in self.fields:
+            raise KeyError(f"Field {checksum_field_name} is not defined")
+        for checksum in self._checksums:
+            if checksum.checksum_field_name == checksum_field_name:
+                checksum.set_checksum_function(checksum_method)
+                return
+        raise KeyError(f"Field {checksum_field_name} has not been defined as a checksum field")
+
+    def _update_checksum_fields(self, field_name: str, checksum_field: str) -> None:
+        for checksum in self._checksums:
+            for mapping in checksum.expr_fields_mapping:
+                if field_name in mapping.dependant_fields:
+                    mapping.evaluated_expression = self.__simplified(copy.copy(mapping.expression))
+
+        arguments: Dict[str, Union[int, Tuple[int, int]]] = {}
+        for checksum in self._checksums:
+            if checksum.checksum_field_name == checksum_field:
+                if not checksum.checksum_func:
+                    raise AttributeError(
+                        f"A callable checksum function must be set in order to "
+                        f"calculate a checksum for {checksum_field}"
+                    )
+                for mapping in checksum.expr_fields_mapping:
+                    if (
+                        isinstance(mapping.evaluated_expression, ValueRange)
+                        and isinstance(mapping.evaluated_expression.lower, Number)
+                        and isinstance(mapping.evaluated_expression.upper, Number)
+                    ):
+                        arguments[str(mapping.expression)] = (
+                            mapping.evaluated_expression.lower.value,
+                            mapping.evaluated_expression.upper.value,
+                        )
+                    elif isinstance(mapping.evaluated_expression, Variable):
+                        if (
+                            mapping.evaluated_expression.name in self.fields
+                            and self._fields[mapping.evaluated_expression.name].set
+                        ):
+                            arguments[str(mapping.expression)] = self._fields[
+                                mapping.evaluated_expression.name
+                            ].typeval.value
+                    elif isinstance(mapping.evaluated_expression, Number):
+                        arguments[str(mapping.expression)] = mapping.evaluated_expression.value
+                if field_name != checksum_field:
+                    self.set(checksum_field, checksum.checksum_func(self.bytestring, **arguments))
```

Comment:
The reason was to prevent a recursion error because `set()` checks if the checksum can be computed (all required information are available) and then calls `calculate checksum()` to do the checksum calculation, which in turn calls `set()` to set the computed checksum to the checksum field. This particular code segment has been removed, but the concept is still present in the current version.